util/accept_server_test.cc

```diff
@@ -276,25 +276,36 @@ TEST_F(AcceptServerTest, Shutdown) {
   constexpr auto kHupMask = POLLHUP;
 #endif
 
-    vector<unique_ptr<FiberSocketBase>> socks;
     constexpr unsigned kNumSocks = 2000;
+
+    vector<unique_ptr<FiberSocketBase>> socks;
+    vector<bool> error_notification(kNumSocks, false);
+
     unsigned called = 0;
+    fb2::CondVarAny cond;
 
     for (unsigned i = 0; i < kNumSocks; ++i) {
       socks.emplace_back(proactor->CreateSocket());
       error_code ec = socks.back()->Connect(ep_);
       EXPECT_FALSE(ec);
 
-      socks.back()->RegisterOnErrorCb([&](int err) {
+      socks.back()->RegisterOnErrorCb([&, i](int err) {
         EXPECT_EQ(err, kHupMask);
-        ++called;
+
+        // MacOs can send multiple notifications per socket.
+        if (!error_notification[i]) {
+          ++called;
+          error_notification[i] = true;
```

Comment:
Unfortunately, this is unsafe (and it is un-intuitive that it is unsafe).
`vector<bool>` is defined in the standard to not be an array of `bool` but rather use a single bit per boolean. It's been considered an unfixable mistake.
https://en.cppreference.com/w/cpp/container/vector_bool
As such, accessing different indices from the different threads may yield undefined behavior...

A common fix is to use `vector<MyEnum>` or `vector<uint8_t>`
(I understand this is just a test, but still, let's make sure it doesn't fail due to this)

Reply:
but I do not access different indices from different threads.
It's all the same thread - it's all wrapped in one huge Await.

Reply:
it would be unsafe even with uint8_t if it were from different thrreads.

Reply:
> but I do not access different indices from different threads. It's all the same thread - it's all wrapped in one huge Await.

Right, I missed that.

> it would be unsafe even with uint8_t if it were from different thrreads.

That's true :)